### PR TITLE
PHPC-827: Update Max Staleness implementation

### PR DESCRIPTION
**Pull request by derickr**

### CODE REVIEW

File: tests/readPreference/readpreference-ctor_error-003.phpt [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 require_once __DIR__ . '/../utils/tools.php';
 
 echo throws(function() {
-    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY, null, ['maxStalenessMS' => 1000]);
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_PRIMARY, null, ['maxStalenessSeconds' => 1000]);
 }, 'MongoDB\Driver\Exception\InvalidArgumentException'), "\n";
 
 echo throws(function() {
-    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY, null, ['maxStalenessMS' => -1]);
+    new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY, null, ['maxStalenessSeconds' => -2]);
```
Comment (jmikola) [inline tests/readPreference/readpreference-ctor_error-003.phpt]: Let's add test cases for `0` (another boundary case) and `42` (something less than 90) as well.

File: tests/manager/manager-ctor-read_preference-error-002.phpt [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
 }, "MongoDB\Driver\Exception\InvalidArgumentException"), "\n";
 
 echo throws(function() {
-    $manager = new MongoDB\Driver\Manager(null, ['maxStalenessMS' => 1231]);
+    $manager = new MongoDB\Driver\Manager(null, ['maxStalenessSeconds' => 1231]);
 }, "MongoDB\Driver\Exception\InvalidArgumentException"), "\n";
 
 echo throws(function() {
-    $manager = new MongoDB\Driver\Manager(null, ['readPreference' => 'secondary', 'maxStalenessMS' => -1]);
+    $manager = new MongoDB\Driver\Manager(null, ['readPreference' => 'secondary', 'maxStalenessSeconds' => -2]);
```
Comment (jmikola) [inline tests/manager/manager-ctor-read_preference-error-002.phpt]: Let's add test cases for `0` (another boundary case) and `42` (something less than 90) as well.

File: src/MongoDB/ReadPreference.c [hunk 3]
```diff
@@ -105,25 +105,25 @@ PHP_METHOD(ReadPreference, __construct)
...
+		if (maxStalenessSeconds != MONGOC_NO_MAX_STALENESS) {
+			if (maxStalenessSeconds < MONGOC_SMALLEST_MAX_STALENESS_SECONDS) {
+				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be >= %d, %" PHONGO_LONG_FORMAT " given", MONGOC_SMALLEST_MAX_STALENESS_SECONDS, maxStalenessSeconds);
+				return;
+			}
+			if (maxStalenessSeconds > INT32_MAX) {
+				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected maxStalenessSeconds to be <= %" PRId32 ", %" PHONGO_LONG_FORMAT " given", INT32_MAX, maxStalenessSeconds);
+				return;
+			}
+			if (maxStalenessSeconds != MONGOC_NO_MAX_STALENESS && mode == MONGOC_READ_PRIMARY) {
```
Comment (jmikola) [inline src/MongoDB/ReadPreference.c]: I just realized this is redundant since we're already within a `maxStalenessSeconds != MONGOC_NO_MAX_STALENESS` conditional. No harm in leaving this in place, though. We can always refactor later.
